File: SDK/CentralService/dotnet/net40/src/CentralService.Service/Errors/CentralServiceException.cs
using System;

namespace CentralService.Service.Errors
{
    /// <summary>
    /// 表示中心服务 SDK 调用失败的运行时异常。
    /// </summary>
    /// <remarks>
    /// 调用方通常应优先读取 <see cref="Error"/> 中的结构化错误上下文，
    /// 而不是只依赖异常消息文本。
    /// </remarks>
    public sealed class CentralServiceException : Exception
    {
        /// <summary>
        /// 使用结构化错误详情构造异常。
        /// </summary>
        /// <param name="error">错误详情，不能为空。</param>
        /// <exception cref="ArgumentNullException">当 <paramref name="error"/> 为 <c>null</c> 时抛出。</exception>
        public CentralServiceException(CentralServiceError error)
            : base(error != null ? error.Message : null)
        {
            if (error == null) throw new ArgumentNullException("error");
            Error = error;
        }

        /// <summary>
        /// 获取结构化错误详情。
        /// </summary>
        public CentralServiceError Error { get; private set; }
    }
}

File: SDK/CentralService/java/service/src/main/java/erp/centralservice/service/errors/CentralServiceException.java
package erp.centralservice.service.errors;

/**
 * 表示中心服务 SDK 调用失败的运行时异常。
 *
 * <p>异常主体是 {@link CentralServiceError}，调用方通常应优先读取
 * {@link #getError()} 获取结构化错误上下文，而不是只依赖异常消息文本。</p>
 */
public final class CentralServiceException extends RuntimeException {
    /** 结构化错误详情。 */
    private final CentralServiceError error;

    /**
     * 使用结构化错误详情构造异常。
     *
     * @param error 错误详情，可为 {@code null}
     */
    public CentralServiceException(CentralServiceError error) {
        super(error != null ? error.message : "CentralServiceException");
        this.error = error;
    }

    /**
     * 返回结构化错误详情。
     *
     * @return 错误详情；如果构造时未提供则为 {@code null}
     */
    public CentralServiceError getError() {
        return error;
    }

    /**
     * 优先输出结构化错误的摘要文本。
     *
     * @return 单行错误描述
     */
    @Override
    public String toString() {
        return error != null ? error.toString() : super.toString();
    }
}


